src/main/java/app/linguistai/bmvp/service/wordbank/WordSelectionService.java

```diff
@@ -0,0 +1,147 @@
+import app.linguistai.bmvp.repository.wordbank.IWordSelectionRepository;
+import org.springframework.stereotype.Service;
+
+import app.linguistai.bmvp.model.User;
+import app.linguistai.bmvp.model.enums.ConfidenceEnum;
+import app.linguistai.bmvp.model.wordbank.UnknownWord;
+import app.linguistai.bmvp.model.wordbank.WordSelection;
+import app.linguistai.bmvp.repository.IAccountRepository;
+import app.linguistai.bmvp.request.QSelectWord;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class WordSelectionService {
+
+    private final IWordSelectionRepository wordSelectionRepository;
+    private final IAccountRepository accountRepository;
+    private final IUnknownWordRepository unknownWordRepository;
+
+    public List<WordSelection> getSelectedWords(QSelectWord selectWord, String email) throws Exception {
+        try {
+            User user = accountRepository.findUserByEmail(email)
+                .orElseThrow(() -> new NotFoundException("User does not exist for given email: [" + email + "]."));
+
+
+            List<String> preservedWords = new ArrayList<>(selectWord.getPreservedWords());
+
+            // Get the previously selected words
+            List<WordSelection> selectedWords = 
+                                wordSelectionRepository
+                                .findByConversationIdAndWordOwnerListUserEmail(
+                                    selectWord.getConversationId(), 
+                                    user.getEmail());
+
+            // If words are not selected yet, select them and save to db
+            if (selectedWords.isEmpty()) {
+                List<UnknownWord> selectedUnknownWords = selectWords(user.getId(), selectWord.getSize());
+
+                // Iterate over each selected word and create a WordSelection object
+                for (UnknownWord unknownWord : selectedUnknownWords) {
+                    WordSelection wordSelection = WordSelection.builder()
+                            .conversationId(selectWord.getConversationId())
+                            .word(unknownWord)
+                            .build();
+
+                    selectedWords.add(wordSelection);
+                }
+
+                // Save the selected words to db
+                wordSelectionRepository.saveAll(selectedWords);
+
+                log.info("New words are selected for user email {}", email);
+            } else {
+                // Remove the words that are not in the preservedWords list from the selectedWords list
+                selectedWords.removeIf(wordSelection -> !preservedWords.contains(wordSelection.getWord().getWord()));
+
+                // Remove all words
+                wordSelectionRepository.deleteAll(selectedWords);
+
+                // If the size of the selectedWords list is less than the required size, select new words
+                if (selectedWords.size() < selectWord.getSize()) {
```

Comment:
Similar to the previous comment, we can invert the condition of the if statement to reduce nesting. However, in this case, I don't think it is that necessary as inverting the condition will introduce duplicate code. It is up to you